target/linux/ar71xx/image/generic.mk

```diff
@@ -229,6 +229,90 @@ define Device/dragino2
 endef
 TARGET_DEVICES += dragino2
 
+define Device/e1700ac-8M
+  DEVICE_TITLE := WHQX E1700AC-8M (8MB flash)
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport kmod-ath10k ath10k-firmware-qca988x
+  BOARDNAME := E1700AC
+  IMAGE_SIZE := 7744k
+  KERNEL_SIZE := 1472k
```

Comment:
This will be a big issue for your boards. In practice, support for your boards might be dropped soon (images won't get generated, have a look here: https://github.com/lede-project/source/commit/f7a6fd31539be54d14d7c52b491b40b26bf8f740). The problem is that our kernel is constantly growing and `1472k` might be not enough very soon (current kernel size is just several KB below this limit).

Are your boards already in MP stage? If not, I strongly suggest you to change mtd organization on all your boards and use `kernel-rootfs` order which would make sure that your boards wouldn't have kernel size issue in future. You would probably need to change `bootm` U-Boot environment variable for that to match different kernel offset.

If your boards are already on the market but your default firmware includes support for U-Boot environment in user space (`fw_printenv`/`fw_setenv` commands), you should be able to support `kernel-rootfs` mtd order by requiring users to change `bootm` variable _before_ installing OpenWrt. Please, have a look for example at `Flash instruction under vendor firmware, using telnet/SSH` section here: https://github.com/lede-project/source/commit/c198ca682c187e61730119134e1dde75da2e7a16.

Also, please decrease code duplication here and use same approach as for [Compex boards](https://github.com/lede-project/source/blob/master/target/linux/ar71xx/image/generic.mk#L917). You can for example include `define Device/e1700ac-8M` in `define Device/e1700ac-16M` definition and just overwrite variables which are different.